pkg/helios/cas/cas_test.go

```diff
@@ -0,0 +1,499 @@
+// Copyright 2025 Oppie Thunder Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// SPDX-License-Identifier: Apache-2.0
+
+package cas
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/good-night-oppie/helios-engine/pkg/helios/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestContentAddressableStorage_BasicOperations tests the fundamental CAS operations
```

Comment:
Tests: subtests here can use t.Parallel() to improve coverage time and catch concurrency issues earlier.